src/parser/cssParser.ts

```diff
@@ -361,16 +361,23 @@ export class Parser {
 	}
 
 	protected _parseRuleSetDeclarationAtStatement(): nodes.Node | null {
-		return this._parseUnknownAtRule();
+		return this._parseMedia(true)
+			|| this._parseSupports(true)
+			|| this._parseLayer(true)
+			|| this._parseUnknownAtRule();
 	}
 
 	public _parseRuleSetDeclaration(): nodes.Node | null {
 		// https://www.w3.org/TR/css-syntax-3/#consume-a-list-of-declarations
 		if (this.peek(TokenType.AtKeyword)) {
 			return this._parseRuleSetDeclarationAtStatement();
 		}
-		return (!this.peek(TokenType.Ident) && this._tryParseRuleset(true))
-			|| this._parseDeclaration();
+
+		if (!this.peek(TokenType.Ident)) {
+			return this._parseRuleset(true);
+		}
+
+		return this._tryParseRuleset(true) || this._parseDeclaration();
```

Comment:
This still isn't fully correct for the CSS specification, but it does work ok in practice.

It is mainly important to have struct rule parsing when the current token isn't an ident. There must not a be a fallback to declaration parsing.

Doing the fallback to declaration parsing causes the wrong auto complete and diagnostics to be shown to users.

----

To be fully correct for the specification it would be better to do this :

1. parse as a declaration
   1. if this passes, return the declaration
   2. if this fails, record the diagnostics and reset the parser
2. parse as a rule
   1. if this passes, return the rule
   2. if this fails, record the diagnostics
3. report all diagnostics

It is unknown if a user is intending a declaration or a rule when both are invalid.

I think it is ok to add this later given that it is a very large refactor for an edge case.